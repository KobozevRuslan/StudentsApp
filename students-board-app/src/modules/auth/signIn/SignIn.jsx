import React from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';

import FormWrapper from '../../main/Components/formWrapper/FormWrapper';

import { signInSchema } from './validation';

import { authLogin } from '../../../store/actions/app';

const SignIn = ({ title, buttonName, inputTypes: { email, password } }) => {
  const dispatch = useDispatch();

  return (
    <FormWrapper
      title={title}
      fields={{ email, password }}
      validation={signInSchema}
      onSubmitHandle={{
        buttonName,
        buttonHandler: values => {
          dispatch(authLogin(values));
        },
      }}
    />
  );
};

SignIn.propTypes = {
  title: PropTypes.string.isRequired,
  buttonName: PropTypes.string.isRequired,
  inputTypes: PropTypes.object.isRequired,
};

export default SignIn;
