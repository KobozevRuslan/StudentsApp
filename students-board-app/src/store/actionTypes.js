export const CHANGE_LOADER = 'CHANGE_LOADER';
export const CHANGE_TAB = 'CHANGE_TAB';
export const AUTH_LOGIN = 'AUTH_LOGIN';
export const AUTH_IN = 'AUTH_IN';
