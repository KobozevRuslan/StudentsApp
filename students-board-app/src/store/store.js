import { applyMiddleware, createStore, combineReducers } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';

import authReducer from './reducers/authReducer';
import appReducer from './reducers/appReducer';
import rootSaga from './sagas/rootSaga';

const rootReducer = combineReducers({
  app: appReducer,
  auth: authReducer,
});

const logger = store => next => action => {
  console.log('action', action);
  let result = next(action);
  console.log('nex state', store.getState());
  return result;
};

const sagaMiddleware = createSagaMiddleware();

export const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(sagaMiddleware, logger)),
);

sagaMiddleware.run(rootSaga);
