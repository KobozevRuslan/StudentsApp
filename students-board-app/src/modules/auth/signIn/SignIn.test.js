import React from 'react';
import { Provider } from 'react-redux';

import { store } from '../../../store/store';

import SignIn from './SignIn';

describe('SignIn component', () => {
  it('should render SignIn component with props', () => {
    const component = shallow(
      <Provider store={store}>
        <SignIn
          title={'Welcome back'}
          link={{ linkName: 'Forgot the password', linkUrl: '/forgotPassword' }}
          buttonName={'Sign In'}
          inputTypes={{ email: '', password: '' }}
        />
      </Provider>,
    );
    console.log(component.debug());
    expect(component).toMatchSnapshot();
  });
});
