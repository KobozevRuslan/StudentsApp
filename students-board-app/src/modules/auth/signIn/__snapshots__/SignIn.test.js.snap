// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn component should render SignIn component with props 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <SignIn
    buttonName="Sign In"
    inputTypes={
      Object {
        "email": "",
        "password": "",
      }
    }
    link={
      Object {
        "linkName": "Forgot the password",
        "linkUrl": "/forgotPassword",
      }
    }
    title="Welcome back"
  />
</ContextProvider>
`;
